ingests/ihme_gbd.py

```diff
@@ -0,0 +1,122 @@
+#    Sex: Both; Male; Female
+#
+# 2d. For gbd_prevalence select the following:
+#
+#    Measure: Prevalence; Incidence
+#    Age: All Ages; Age-standardized; Under 5; 5-14 years; 15-49 years; 50-69 years; 70+ years
+#    Metric: Number; Rate; Percent
+#    Year: select all
+#    Cause: select all
+#    Context: Cause
+#    Location: select all countries
+#    Sex: Both
+#
+# 2e. For gbd_risk select the following:
+#
+#    Measure: Deaths; DALYs (Disability-Adjusted Life Years)
+#    Age: All Ages; Age-standardized; Under 5; 5-14 years; 15-49 years; 50-69 years; 70+ years
+#    Metric: Number; Percent; Rate
+#    Year: select all
+#    Risks: select all
+#    Cause: Total All Causes; Cardiovascular diseases; Lower respiratory infections; Diarrheal diseases; Neoplasms
+#    Context: Risk
+#    Location: select all countries
+#    Sex: Both
+#
+# 3. Wait for the data to be prepared and for a download link to be sent
+import glob
+import os
+
+import click
+import pandas as pd
+from structlog import get_logger
+
+from owid.walden import add_to_catalog
+
+LOCAL_PATH = "/Users/fionaspooner/Documents/temp/gbd_2021/"
+
+log = get_logger()
+
+
+@click.command()
+@click.option(
+    "--upload/--skip-upload",
+    default=True,
+    type=bool,
+    help="Upload dataset to Walden",
+)
+def main(upload: bool) -> None:
+    names = ["gbd_cause", "gbd_risk", "gbd_prevalence", "gbd_child_mortality", "gbd_mental_health"]
+    descriptions = ["Deaths and DALYs", "Risk factors", "Prevalence and incidence", "Child mortality", "Mental health"]
+    for name, description in zip(names, descriptions):
+        log.info("Combining data for:", df_name=name)
+        outpath = combine_csvs(name=name, inpath=f"{LOCAL_PATH}{name}/csv/")
+        metadata = {
+            "namespace": "ihme_gbd",
+            "short_name": name,
+            "name": "Institute for Health Metrics and Evaluation - Global Burden of Disease (2019)",
+            "description": description,
+            "publication_year": 2019,
+            "source_name": "Institute for Health Metrics and Evaluation - Global Burden of Disease Collaborative Network",
+            "url": "https://vizhub.healthdata.org/gbd-results/",
+            "source_data_url": "https://unstats.un.org/sdgapi/swagger/",
+            "file_extension": "feather",
+            "license_url": "https://www.healthdata.org/data-tools-practices/data-practices/terms-and-conditions",
+        }
+        log.info("Adding data to catalog:", df_name=name)
+        add_to_catalog(metadata=metadata, filename=outpath, upload=upload, public=False)
+
+
+def combine_csvs(name: str, inpath: str) -> str:
+    # setting the path for joining multiple files
+    files = os.path.join(inpath, "*.csv")
+    # list of merged files returned
+    files = glob.glob(files)
+    # joining files with concat and read_csv
+    df = pd.concat(map(pd.read_csv, files), ignore_index=True)
+    outpath = f"{inpath}{name}.feather"
+    df.to_feather(outpath)
```

Comment:
15GB sounds like a lot, could you try repacking it before saving to feather with
```
from owid.catalog.frames import repack_frame

df = repack_frame(df)
df.to_feather(outpath)
```
to get more compact types? (like categoricals etc.)

Reply:
Ah wow - now it's more like 3.5GB! Will upload now.